src/Microsoft.AspNet.FileProviders.Physical/PhysicalFileProvider.cs

```diff
@@ -45,12 +45,20 @@ public PhysicalFileProvider(string root)
 
         private string GetFullPath(string path)
         {
-            var fullPath = Path.GetFullPath(Path.Combine(Root, path));
-            if (!IsUnderneathRoot(fullPath))
+            try
+            {
+                var fullPath = Path.GetFullPath(Path.Combine(Root, path));
+                if (!IsUnderneathRoot(fullPath))
+                {
+                    return null;
+                }
+                return fullPath;
+            }
+            catch (NotSupportedException)
             {
-                return null;
+                // path contains a colon (":") that is not part of a volume identifier (for example, "c:\").
```

Comment:
We should avoid first chance exceptions like these in cases where we know it will throw.